checkov/sca_package/scanner.py

```diff
@@ -98,3 +100,17 @@ def parse_api_result(self, origin_file_path: Path, response: str) -> dict:
         raw_result = json.loads(decompress_file_gzip_base64(response))
         raw_result['repository'] = str(origin_file_path)
         return raw_result
+
+    def request_wrapper(self, method: str, url: str, headers: Any, data: Optional[Any] = None):
+        # using of "retry" mechanism for 'requests.request' due to unpredictable 'ConnectionError' that appears
+        # from time to time ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer')).
+        remaining_tries = self._request_max_tries
+        while True:
+            try:
+                return requests.request(method, url, headers=headers, data=data)
+            except requests.exceptions.ConnectionError as err:
+                remaining_tries -= 1
+                if remaining_tries == 0:
+                    raise err
+                else:
+                    time.sleep(self._sleep_between_request_tries * (self._request_max_tries - remaining_tries))
```

Comment:
@itai1357 Let's also catch and retry 504 HTTP errors:
```
HTTPError - 504 Server Error: Gateway Time-out for url: https://www.bridgecrew.cloud/api/v1/vulnerabilities/scan
```
Also - let's log an info saying we got the error ..., retrying

Reply:
@nimrodkor resolved